ern/index.go

```diff
@@ -123,70 +123,75 @@ func isUniqueErr(err error) bool {
 	return e.Code == sqlite3.ErrConstraint && e.ExtendedCode == sqlite3.ErrConstraintUnique
 }
 
-// indexMessageHeader indexes an ERN MessageHeader based on its MessageId,
-// MessageThreadId, MessageSender, MessageRecipient and MessageCreatedDateTime.
-func (i *Indexer) indexMessageHeader(ernID *cid.Cid, obj *meta.Object) error {
-	graph := meta.NewGraph(i.store, obj)
+ // decode decodes whatever is stored at path into the given value
+func DecodeIndexObj(i *Indexer, metaObj *meta.Object, v interface{}, path ...string) (err error) {
```

Comment:
DecodeIndexObj decodes...

Reply:
What about func (i*Index) Decode(...

Reply:
Can you clarify what you mean? I am not sure if i follow, sorry,